src/jobs/microk8s_install.yml

```diff
@@ -124,24 +130,88 @@ steps:
         END_VALUES
 
   - run:
-      name: Get chart and setup codacy helm repos
+      name: Setup helm repos
       command: |
         set -x
-        git clone https://github.com/codacy/chart.git
-        cd chart/
-        git checkout << parameters.chart_branch >>
-
         sudo helm repo add codacy-stable    https://charts.codacy.com/stable
         sudo helm repo add codacy-unstable  https://charts.codacy.com/unstable
         sudo helm repo add codacy-incubator https://charts.codacy.com/incubator
         sudo helm repo add codacy-external  https://charts.codacy.com/external
 
-  - deploy:
-      name: Install Codacy to microk8s
+  - when:
+      condition: << parameters.clone_chart_source >>
+      steps:
+        - run:
+            name: Get codacy chart
+            command: |
+              set -x
+              git clone https://github.com/codacy/chart.git
+              cd << parameters.chart_dir_path >>
+              git checkout << parameters.chart_branch >>
+
+  - run:
+      name: Add namespace and docker secret
+      command: |
+        set -x
+        if [[ -z "$(echo $PATH | grep snap)" ]] ; then export PATH=$PATH:/snap/bin ; fi
+
+        sudo microk8s.kubectl create namespace codacy
+        sudo microk8s.kubectl create secret docker-registry docker-credentials \
+          --docker-username=$DOCKER_USERNAME --docker-password=$DOCKER_PASSWORD -n codacy
+
+  - run:
+      name: Run additional pre-install commands (if any)
+      command: |
+        set -x
+        if [[ -z "$(echo $PATH | grep snap)" ]] ; then export PATH=$PATH:/snap/bin ; fi
+        << parameters.extra_preinstall_commands >>
+
+  - when:
+      condition: << parameters.use_chart_source >>
```

Comment:
Not sure I understand why we have this distinction ... can you explain?

Reply:
This was to support installing from the chart source instead as from the repo, but it is not needed any more.